src/Hedgehog/Gen.fs

```diff
@@ -179,7 +183,7 @@ module Gen =
         if Array.isEmpty xs then
             return crashEmpty "xs"
         else
-            let! ix = integral <| Range.constant 0 (Array.length xs - 1)
+            let! ix = integral (Range.constant 0 (Array.length xs - 1))
```

Comment:
```suggestion
            let! ix = Array.length xs - 1 |> Range.constant 0 |> integral
```

Reply:
I'm not sure I like this piping, since arithmetic is being used before the pipe. Maybe that sub-expression can be given a name?

Reply:
Not a big deal.  You can leave it as it.

Reply:
I made the range construction part of the range module, under a function called `Range.ofArray`. This function just creates a `constant 0 (Array.length xs - 1)` as a helper, but it also cleaned up this part entirely. With the added bonus of removing duplication. :+1: 

The new function is marked `internal` until we decide to expose it.